libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/Pkcs11KeyConfig.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.ssl;
+
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * A {@link SslKeyConfig} that builds a Key Manager from a keystore file.
+ */
+public class Pkcs11KeyConfig extends SslKeystoreConfig {
```

Comment:
Not sure whether we want this - we don't have tests for it, and we talked about dropping PKCS#11 support.
But for now it's here since X-Pack has it.

Reply:
I'm going to drop PKCS#11 support from this PR, and open a new PR to remove PKCS#11 support from X-Pack (in 8.0 only).
That way we have an explicit PR with a breaking change, and these PRs can remain non-issue PRs.